File: VehicleReservationClient/src/app/_modules/VehicleForUserReservationDto.ts
import { VehicleType } from "./enums";

export interface VehicleForUserReservationDto {
  vehicleId: string;
  name: string;
  type: VehicleType;
  mark: string;
  seats: number;
  fuel: string;
  year: number;
  cost: number;
  registrationNumber: string;
}

File: VehicleReservationAPI/DTOs/VehicleForUserReservationDto.cs
using VehicleReservationAPI.Enums;

namespace VehicleReservationAPI.DTOs
{
    public record VehicleForUserReservationDto
    {
        public required string Name { get; init; }
        public required VehicleType Type { get; init; }
        public required string Mark { get; init; }
        public required int Seats { get; init; }
        public required string Fuel { get; init; }
        public required int Year { get; init; }
        public required double Cost { get; init; }
        public required string RegistrationNumber { get; init; }
    }
}